buildSrc/src/main/kotlin/org/gradle/plugins/publish/PublishPublicLibrariesPlugin.kt

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradle.plugins.publish
+
+import accessors.base
+import accessors.java
+import accessors.groovy
+import org.gradle.api.Action
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.artifacts.ExternalDependency
+import org.gradle.api.artifacts.ProjectDependency
+import org.gradle.api.artifacts.maven.Conf2ScopeMappingContainer
+import org.gradle.api.artifacts.maven.MavenResolver
+import org.gradle.api.execution.TaskExecutionGraph
+import org.gradle.api.internal.artifacts.publish.DefaultPublishArtifact
+import org.gradle.api.tasks.Upload
+import org.gradle.api.tasks.bundling.Jar
+import org.gradle.kotlin.dsl.*
+import java.io.File
+import java.util.*
+
+open class PublishPublicLibrariesPlugin : Plugin<Project> {
+
+    override
+    fun apply(project: Project): Unit = project.run {
+        apply {
+            plugin("maven")
+        }
+
+        val publishCompile by configurations.creating
+        val publishRuntime by configurations.creating
+        val compile by configurations.getting {
+            extendsFrom(publishCompile)
+        }
+        lateinit var pomFile: File
+
+        val generatePom by tasks.creating {
+            pomFile = File(temporaryDir, "pom.xml")
+            extra.set("pomFile", pomFile)
+
+            doLast {
+                dependencies {
+                    publishCompile.allDependencies.withType<ProjectDependency>().forEach {
+                        publishRuntime("org.gradle:${it.dependencyProject.base.archivesBaseName}:${version}")
+                    }
+                    publishCompile.allDependencies.withType<ExternalDependency>().forEach {
+                        publishRuntime(it)
+                    }
+                }
+
+                val install by tasks.getting(Upload::class)
+                val mavenInstaller by install.repositories
```

Comment:
:x: `mavenInstaller` is a property of `MavenRepositoryHandlerConvention`.

Something like the following would do:
```kotlin
install.repositories {
  withConvention(MavenRepositoryHandlerConvention::class) {
    mavenInstaller {
      pom.scopeMappings.mappings.clear()
      ...
    }
  }
}
```

Reply:
Very nice. I have applied this.